scaleway/resource_container_test.go

```diff
@@ -0,0 +1,392 @@
+		})
+
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckScalewayContainerDestroy(tt *TestTools) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+		for _, rs := range state.RootModule().Resources {
+			if rs.Type != "scaleway_container_namespace" {
+				continue
+			}
+
+			api, region, id, err := containerAPIWithRegionAndID(tt.Meta, rs.Primary.ID)
+			if err != nil {
+				return err
+			}
+
+			_, err = api.DeleteContainer(&container.DeleteContainerRequest{
+				ContainerID: id,
+				Region:      region,
+			})
+
+			if err == nil {
+				return fmt.Errorf("container (%s) still exists", rs.Primary.ID)
+			}
+
+			if !is404Error(err) {
+				return err
+			}
+		}
+
+		return nil
+	}
+}
+
+func testRemoveAllRegistryNamespaceGenerated(tt *TestTools) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		return sweepRegions([]scw.Region{scw.RegionFrPar, scw.RegionNlAms}, func(scwClient *scw.Client, region scw.Region) error {
+			meta := tt.Meta
+			registryAPI := registry.NewAPI(meta.scwClient)
+			listNamespaces, err := registryAPI.ListNamespaces(
+				&registry.ListNamespacesRequest{Region: region}, scw.WithAllPages())
+			if err != nil {
+				return fmt.Errorf("error listing namespaces in (%s) in test: %s", region, err)
+			}
+
+			for _, ns := range listNamespaces.Namespaces {
+				_, err := registryAPI.DeleteNamespace(&registry.DeleteNamespaceRequest{
+					NamespaceID: ns.ID,
+					Region:      region,
+				})
+				if err != nil {
+					return fmt.Errorf("error deleting namespace in sweeper: %s", err)
+				}
+			}
+
+			return nil
+		})
+	}
+}
+
+func testAddImageToRegistryScaleway(tt *TestTools, n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("not found: %s", n)
+		}
+		api, region, id, err := registryAPIWithRegionAndID(tt.Meta, rs.Primary.ID)
+		if err != nil {
+			return nil
+		}
+
+		ns, err := api.WaitForNamespace(&registry.WaitForNamespaceRequest{
+			NamespaceID: id,
+			Region:      region,
+		})
+		if err != nil {
+			return fmt.Errorf("error waiting namespace: %v", err)
+		}
+
+		meta := tt.Meta
+		var errorMessage ErrorRegistryMessage
+
+		accessKey, _ := meta.scwClient.GetAccessKey()
+		secretKey, _ := meta.scwClient.GetSecretKey()
+		authConfig := types.AuthConfig{
+			ServerAddress: ns.Endpoint,
+			Username:      accessKey,
+			Password:      secretKey,
+		}
+
+		cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+		if err != nil {
+			return fmt.Errorf("could not connect to Docker: %v", err)
+		}
+
+		encodedJSON, err := json.Marshal(authConfig)
+		if err != nil {
+			return fmt.Errorf("could not marshal auth config: %v", err)
+		}
+
+		ctx := context.Background()
+		authStr := base64.URLEncoding.EncodeToString(encodedJSON)
+
+		out, err := cli.ImagePull(ctx, "docker.io/library/alpine", types.ImagePullOptions{})
```

Comment:
Could you introduce a variable for this image? Maybe we want to change it later.